src/ray/gcs/gcs_server/gcs_health_check_manager.cc

```diff
@@ -132,13 +168,13 @@ void GcsHealthCheckManager::HealthCheckContext::Stop() { stopped_ = true; }
 void GcsHealthCheckManager::AddNode(const NodeID &node_id,
                                     std::shared_ptr<grpc::Channel> channel) {
   io_service_.dispatch(
```

Comment:
since this dispatches we don't need it to be run on the same thread. do we have a rule about which method is thread safe, which one must be called on same thread? need to doc them out

Reply:
Sounds good, add documentation for all three public functions.